File: Data Structures & Algorithms/eating-bananas/submission-6.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        
        l = 1
        r = max(piles)

        while l < r:
            k = l + ( (r - l) // 2 )

            hours = 0
            for b in piles:
                hours += (b + k - 1) // k
            if hours <= h:
                r = k
            else:
                l = k + 1
        return l
            
            

File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        unordered_map<int, int> seen;

        int bignum;
        int lownum;

        for (int i=0; i < numbers.size(); i++) {
            int cur = numbers[i];
            seen[cur] = i;
        }

        for (int i=0; i < numbers.size(); i++) {
            int curnum = numbers[i];

            int offset = target - curnum;

            if (seen.count(offset)) {

                if (seen[offset] != i) {
                    bignum = max((seen[offset])+1, i+1);
                    lownum = min((seen[offset])+1, i+1);

                    return {lownum, bignum};
                }
            }
        }
        return {lownum, bignum};
    }
};
